### PR TITLE
tso: only run Local TSO related daemons when EnableLocalTSO

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Only run Local TSO related daemons when `EnableLocalTSO` is true.

### What is changed and how it works?

Check `EnableLocalTSO` flag before running these daemons.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

File: server/tso/allocator_manager.go [hunk 1]
```diff
@@ -533,24 +533,38 @@ func (am *AllocatorManager) campaignAllocatorLeader(
...
+					return
+				}
+			}
+		}()
+	}
+
+	// Common daemon goroutine.
 	tsTicker := time.NewTicker(am.updatePhysicalInterval)
 	defer tsTicker.Stop()
-	patrolTicker := time.NewTicker(patrolStep)
```
Comment (HunDunDM) [inline server/tso/allocator_manager.go:538]:
```go
	var patrolTickerC, checkerTickerC <- chan time.Time
	if am.enableLocalTSO {
		patrolTicker := time.NewTicker(patrolStep)
		patrolTickerC = patrolTicker.C
		defer patrolTicker.Stop()
		
		checkerTicker := time.NewTicker(PriorityCheck)
		checkerTickerC = checkerTicker.C
		defer checkerTicker.Stop()
	}
```
nil chan will always block.

**Review by HunDunDM — APPROVED** [review]

File: server/tso/allocator_manager.go [hunk 2]
```diff
@@ -535,18 +535,25 @@ func (am *AllocatorManager) campaignAllocatorLeader(
...
-	defer patrolTicker.Stop()
-	checkerTicker := time.NewTicker(PriorityCheck)
-	defer checkerTicker.Stop()
+	var patrolTickerC, checkerTickerC <-chan time.Time
+	// Local TSO related daemon goroutines only work when enableLocalTSO is true.
+	if am.enableLocalTSO {
+		patrolTicker := time.NewTicker(patrolStep)
```
Comment (rleungx) [inline server/tso/allocator_manager.go:541]: `patrolTickerC = time.NewTicker(patrolStep)`?

Comment (JmPotato) [reply server/tso/allocator_manager.go:541]: It's `patrolTickerC = patrolTicker.C` in line #542.

File: server/tso/allocator_manager.go [hunk 3]
```diff
@@ -1081,8 +1084,15 @@ func (am *AllocatorManager) GetMaxLocalTSO(ctx context.Context) (*pdpb.Timestamp
 		}
 	}
 	maxTSO := &pdpb.Timestamp{}
+	if len(clusterDCLocations) == 0 {
```
Comment (rleungx) [inline server/tso/allocator_manager.go:1087]: move it before line 1080?

**Review by rleungx — APPROVED** [review]
The rest LGTM.

File: server/tso/allocator_manager.go (hunk 3 above)
Comment (JmPotato) [reply server/tso/allocator_manager.go]: The loop above may change its length. So we should check after the loop.

**Pull request merged**